omnigibson/action_primitives/curobo.py

```diff
@@ -225,119 +291,155 @@ def update_obstacles(self, ignore_paths=None):
                 be ignored when updating obstacles
         """
         print("Updating CuRobo world, reading w.r.t.", self.robot.prim_path)
+        emb_sel = CuroboEmbodimentSelection.DEFAULT  # all embodiment selections share the same world collision checker
+
         ignore_paths = [] if ignore_paths is None else ignore_paths
 
         # Ignore any visual only objects and any objects not part of the robot's current scene
         ignore_scenes = [scene.prim_path for scene in og.sim.scenes]
         del ignore_scenes[self.robot.scene.idx]
         ignore_visual_only = [obj.prim_path for obj in self.robot.scene.objects if obj.visual_only]
 
-        # Filter out any objects corresponding to ground
-        ground_paths = {obj.prim_path for obj in self.robot.scene.objects if obj.category in GROUND_CATEGORIES}
-
-        obstacles = self._usd_help.get_obstacles_from_stage(
+        obstacles = get_obstacles(
             reference_prim_path=self.robot.root_link.prim_path,
             ignore_substring=[
                 self.robot.prim_path,  # Don't include robot paths
                 "/curobo",  # Don't include curobo prim
                 "visual",  # Don't include any visuals
-                "ground_plane",  # Don't include ground plane
-                *ground_paths,  # Don't include collisions with any ground-related objects
                 *METALINK_PREFIXES,  # Don't include any metalinks
                 *ignore_scenes,  # Don't include any scenes the robot is not in
                 *ignore_visual_only,  # Don't include any visual-only objects
                 *ignore_paths,  # Don't include any additional specified paths
             ],
-        ).get_collision_check_world()
-        self.mg.update_world(obstacles)
+        )
+        self.mg[emb_sel].update_world(obstacles)
```

Comment:
should we just set `CuroboEmbodimentSelection.DEFAULT` directly here?

Reply:
Done.